### PR TITLE
Improve performance of Maven explorer

### PR DESCRIPTION
**Pull request by Eskibear**
1. Now effective-pom is not calculated until the user expands the "Plugins" node. 
2. When the "Plugins" node is expanded, plugin nodes are rendered with the qualified names. A corresponding mvn command to fetch plugin info is queued to execute for each plugin. After acquiring the info (prefix), it refreshes the corresponding node, putting the prefix ahead.

### CODE REVIEW

File: src/taskExecutor.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class Queue<T> {
```
Comment (jdneo) [inline src/taskExecutor.ts]: license

File: src/taskExecutor.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        this._store.push(val);
+    }
+    public pop(): T | undefined {
+        return this._store.shift();
+    }
+}
+
+class TaskExecutor {
+    private _tasks: Queue<any> = new Queue();
+    private _executing: boolean = false;
```
Comment (jdneo) [inline src/taskExecutor.ts]: isExecuting

File: src/taskExecutor.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public push(val: T): void {
+        this._store.push(val);
+    }
+    public pop(): T | undefined {
+        return this._store.shift();
+    }
+}
+
+class TaskExecutor {
+    private _tasks: Queue<any> = new Queue();
```
Comment (jdneo) [inline src/taskExecutor.ts]: Add `dispose()` function which clear the queue when deactivating

File: src/taskExecutor.ts (hunk 1 above)
Comment (Eskibear) [reply src/taskExecutor.ts]: added

File: src/taskExecutor.ts (hunk 3 above)
Comment (Eskibear) [reply src/taskExecutor.ts]: added.

**Pull request merged**